helion/_compiler/inductor_lowering.py

```diff
@@ -699,8 +810,31 @@ def index_expr(self, expr: sympy.Expr, dtype: torch.dtype) -> str:
         name = self.cg.lift(
             expr_from_string(self.cg.device_function.user_sympy_expr(expr))
         ).id
+
+        # If the lifted symbol refers to a `tl.constexpr` kernel
+        # argument (for example a tile/block size constant such as
+        # `_BLOCK_SIZE_1`) the resulting Triton value is not a tensor
+        # and therefore does not expose a `.to` method.
+        if name in self.cg.device_function._constexpr_args:
+            return name
+
         return f"{name}.to({triton_type(dtype)})"
 
+    def truediv(self, a: str, b: str) -> str:  # pyre-ignore[14]
+        """Override truediv to use unpadded dimension size instead of padded size when appropriate."""
+        from .compile_environment import CompileEnvironment
+
+        if isinstance(b, str) and b.startswith("_BLOCK_SIZE_"):
+            block_idx = int(b.split("_BLOCK_SIZE_")[-1])
+            block_info = CompileEnvironment.current().block_sizes[block_idx]
+            if block_info.is_padded():
+                # This block size was padded, use the unpadded size instead
+                unpadded_size_var = f"_UNPADDED_SIZE_{block_idx}"
+                if unpadded_size_var in self.cg.device_function._constexpr_args:
+                    return self.parent_handler.truediv(a, unpadded_size_var)
+
+        return self.parent_handler.truediv(a, b)
+
```

Comment:
I believe this will be incorrect if we are operating on multiple tiles.

If we break 100 into tiles of 64 and 36 then the divisor will not be a constant.

This case is already handled for reduction dimensions, so I'd suggest changing this to use rdim semantics.